src/network/netdev/netdev.c

```diff
@@ -790,7 +792,11 @@ int netdev_load_one(Manager *manager, const char *filename) {
                 .state = _NETDEV_STATE_INVALID, /* an invalid state means done() of the implementation won't be called on destruction */
         };
 
-        dropin_dirname = strjoina(basename(filename), ".d");
+        r = path_extract_filename(filename, &file_basename);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract basename of '%s', ignoring: %m", filename);
```

Comment:
you are not ignoring anything here, but failing. Hence, drop the `, ignoring` suffix